### ISSUE TITLE
Why using a set?

### ISSUE THREAD

**Issue by pulgamecanica** [asarandi/advent-of-code#1]
Hello, I'm doing AOC (42 student)
I like to see how other people solve it. Your solution seamed very clean but I have one question:
Why did you decide to use a set? (A set will exclude elements which are equal)
This will cause an issue if your input is lets say:
```
.42.
..*.
..42
```

Your code will return 42, and the result should be 84 I believe, maybe I missed something on the instructions?

Perhaps you were lucky the input didn't have any repeated numbers?

https://github.com/asarandi/advent-of-code/blob/5325d8f278786298579c1a3a41e893be931263c9/2023-python/day_03/script.py#L24

**Comment by asarandi** [comment]
Hi,
Thank you for the question,
Yes, your observation is correct:

I used a `set` to avoid duplicates because I am iterating over special symbols `!@#$%^&`
and for each one of those i am checking all **8 adjacent cells** for numbers (horizontal, vertical and diagonal).

```
123
 $
456
```

In this example the `$` would match each number three times (because of diagonals).
In your example the asterisk symbol `*` would match each `42` twice.

So yes, my solution would not work for input like your example,
and yes, my input did not have any repeated numbers adjacent to same symbol.
(it does have repeated numbers but all of them are more than 2 lines apart).

**Comment by pulgamecanica** [comment]
Yeah, I understood why you want a set, but I feel this solution could be improved, if for each number founded you save for example, the index where it starts, and add that to your set, so you allow same numbers, with different positions.
And for neighbors sharing the same number, it wont be repeated, because both the number and index would be shared.

**Comment by asarandi** [comment]
I'd like to keep the code short and simple and avoid code bloat.
That extra code is not needed (because there is no such input).

**Comment by pulgamecanica** [comment]
I guess there is a possibly someone gets such an input... you got lucky 👀 haha jkjk
It is going extra, but I believe it's important to at least mention it in an issue (which we've done), in case someone has the same question!
Cheers & keep it up!
GL for tomorrow exercise! 

42 Rocks!
